controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -621,106 +632,130 @@ def _prepare_mocks_for_get_host_by_identifiers_(self, result_reader_iter, hosts)
+        self.assertEqual('test_host_s', hostname)
+        self.assertEqual([config.NVME_OVER_FC_CONNECTIVITY_TYPE], connectivity_types)
 
     @patch("controller.array_action.svc_cli_result_reader.SVCListResultsReader.__iter__")
     def test_get_host_by_identifiers_return_fc_host(self, result_reader_iter):
-        host_1 = self._get_host_as_dictionary('host_id_1', 'test_host_1', [], ['abc1'])
-        host_2 = self._get_host_as_dictionary('host_id_2', 'test_host_2', [''], ['abc2'])
-        host_3 = self._get_host_as_dictionary('host_id_3', 'test_host_3', ['iqn.test.3'], ['abc1', 'abc3'])
+        host_1 = self._get_host_as_dictionary('host_id_1', 'test_host_1', wwpns_list=['wwn1'], iscsi_names_list=[])
+        host_2 = self._get_host_as_dictionary('host_id_2', 'test_host_2', wwpns_list=['wwn2'], iscsi_names_list=[''])
+        host_3 = self._get_host_as_dictionary('host_id_3', 'test_host_3', wwpns_list=['wwn3', 'wwn4'],
+                                              iscsi_names_list=['iqn.test.3'])
         hosts = [host_1, host_2, host_3]
-        self._prepare_mocks_for_get_host_by_identifiers_(result_reader_iter, hosts)
-        host, connectivity_type = self.svc.get_host_by_host_identifiers(Initiators(
-            'iqn.test.6', ['abc3', 'ABC1']))
-        self.assertEqual('test_host_3', host)
-        self.assertEqual([config.FC_CONNECTIVITY_TYPE], connectivity_type)
+        self._prepare_mocks_for_get_host_by_identifiers(result_reader_iter)
+        hostname, connectivity_types = self.svc.get_host_by_host_identifiers(
+            Initiators('Test_nqn', ['wwn4', 'WWN3'], 'iqn.test.6'))
+        self.assertEqual('test_host_3', hostname)
+        self.assertEqual([config.FC_CONNECTIVITY_TYPE], connectivity_types)
 
         result_reader_iter.return_value = self._get_detailed_hosts_list_result(hosts)
-        host, connectivity_type = self.svc.get_host_by_host_identifiers(Initiators(
-            'iqn.test.6', ['abc3']))
-        self.assertEqual('test_host_3', host)
-        self.assertEqual([config.FC_CONNECTIVITY_TYPE], connectivity_type)
+        hostname, connectivity_types = self.svc.get_host_by_host_identifiers(
+            Initiators('Test_nqn', ['wwn3'], 'iqn.test.6'))
+        self.assertEqual('test_host_3', hostname)
+        self.assertEqual([config.FC_CONNECTIVITY_TYPE], connectivity_types)
+
+    @patch("controller.array_action.svc_cli_result_reader.SVCListResultsReader.__iter__")
+    def test_get_host_by_identifiers_no_other_ports_return_fc_host(self, result_reader_iter):
+        host_with_iqn_list = self._get_host_as_dictionary('host_id_s', 'test_host_s', wwpns_list=['WWNs'])
+        self._prepare_mocks_for_get_host_by_identifiers(result_reader_iter, special_host=host_with_iqn_list)
+        hostname, connectivity_types = self.svc.get_host_by_host_identifiers(
+            Initiators('Test_nqn', ['Test_wwn', 'WWNs'], 'Test_iqn'))
+        self.assertEqual('test_host_s', hostname)
```

Comment:
is there a meaning to the `s` in `test_host_s` here?

Reply:
so could we have an indicative name? e.g. `test_special_host`

Reply:
I tried to keep aligned to the other names - test_host_1, test_host_2, test_host_s ...

Reply:
though I would prefer a non-festive term, e.g. "custom"

Reply:
its not "custom", its fields are the same, just not one of 1-3.. 
will it be better if I change the `s` to `0`?

Reply:
> its fields are the same

I thought the point of this special host is to have custom field **values**.
e.g. a "custom" `wwpns_list=['WWNs']` instead of the default `wwpns_list=['wwn3']`

I'm not sure `0` is clearer

Reply:
ok, done

Reply:
could we use "custom" instead of "special" in the parameter name as well?

Reply:
done